src/offspot_demo/setup.py

```diff
@@ -46,92 +60,153 @@ def install_symlink():
 
     Will be switched between maint and prod, but for now point to maint
     """
-    print("Installing docker-compose symlink")
+    logger.info("Installing docker-compose symlink")
     DOCKER_COMPOSE_SYMLINK_PATH.unlink(missing_ok=True)
     DOCKER_COMPOSE_SYMLINK_PATH.symlink_to(DOCKER_COMPOSE_MAINT_PATH)
 
 
-def check_systemd_service(
-    ok_return_codes: list[int] | None = None,
+def setup_check_systemd_service(
+    unit_fullname: str,
     *,
     check_running: bool = False,
+    check_waiting: bool = False,
     check_enabled: bool = False,
-) -> subprocess.CompletedProcess[str]:
+) -> None:
     """Check status of the systemd unit
 
     By default, check at least that the unit is loaded properly (i.e. parsing is ok)
     If check_running is True, it also checks that the unit is running
     If check_enabled is True, it also checks that the unit is enabled
+    If check_waiting is True, it also checks that the unit is waiting
     """
-    status = run_command(
-        ["systemctl", "status", "--no-pager", f"{SYSTEMD_OFFSPOT_UNIT_NAME}.service"],
-        ok_return_codes=ok_return_codes,
-    )
-    if "Loaded: loaded" not in status.stdout:
-        print("systemd unit not loaded properly:")
-        print(status.stdout)
+
+    try:
+        check_systemd_service(
+            unit_fullname=unit_fullname,
+            check_running=check_running,
+            check_enabled=check_enabled,
+            check_waiting=check_waiting,
+        )
+    except SystemdNotLoadedError as exc:
+        logger.error(f"systemd unit not loaded properly:\n{exc.stdout}")
```

Comment:
Seems good enough to me:

```suggestion
    except (SystemdNotLoadedError,  SystemdNotRunningError , SystemdNotWaitingError, SystemdNotEnabledError) as exc:
        logger.exception(exc)
        sys.exit(exc.code)  # ?
```

Logging using `.exception()` has the benefit of only showing on DEBUG level 

```suggestion
        logger.error(f"systemd unit not loaded properly")
        logger.exception(exc)
```

You could even pass it the stdout string directly

```suggestion
        logger.error(f"systemd unit not loaded properly")
        logger.exception(exc.stdout)
```

Reply:
> Logging using .exception() has the benefit of only showing on DEBUG level

I don't think so, the log level of .exception() is ERROR if I'm not mistaken. But I can definitely use logger.debug(...)

Reply:
Or in fact, here we probably always want the exception details

Reply:
Indeed it's `ERROR`. I guess I understand why.